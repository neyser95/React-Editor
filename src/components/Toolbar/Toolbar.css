/* * Block */
.Toolbar{
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  padding: 10px;
  background-color: grey;
  border: 1px solid rgba(0, 0, 0, 0.5);
  box-sizing: border-box;
}
/* * Button Element*/
.Toolbar__btn{
  padding: 5px;
  margin: 2px 5px;
  box-sizing: border-box;
  border: 1px solid rgb(0, 0, 0);
  border-radius: 5px;
  cursor: pointer;
  width: 30px;
  height: 30px;
}
/* * Button Modifiers */
.Toolbar__btn--active{
  color: white;
  background-color: rgb(0, 0, 0);
}
.Toolbar__btn:hover{
  box-shadow: 1px 1px 1px black;
}