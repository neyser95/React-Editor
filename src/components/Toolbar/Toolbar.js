import React from 'react';
import classNames from './Toolbar.css';

// * Made toolBar a functional component and will only handle rendering of the toolbar.
const toolBar = (props) => {
  const {toolbarButtons} = props;

  const buttons = Object.keys(props.toolbarButtons).map((button, i) => {
    return (
      <button key={button + i} className={classNames.Toolbar__btn}>
        <i className={toolbarButtons[button].icon}></i>
      </button>
    );
  });

  return (
    <div className={classNames.Toolbar}>
      {buttons}
    </div>
  );
}

export default toolBar;