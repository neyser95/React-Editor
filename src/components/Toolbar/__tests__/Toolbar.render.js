import React from 'react';
import renderer from 'react-test-renderer';
import Toolbar from '../Toolbar';

const state = {
  toolbarButtons: {
    underline: {
      icon: 'fas fa-underline',
      event: function () {
        document.execCommand('underline', false, '');
      }
    },
    italic: {
      icon: 'fas fa-italic',
      event: function () {
        document.execCommand('italic', false, '');
      }
    }
  }
};

//* Tests to make sure that comopnent does not change unexpectdly.
test('Toolbar will render will all the underline and italic buttons passed as props', () => {
  const toolbar = renderer.create(
    <Toolbar toolbarButtons={state.toolbarButtons}/>
  );

  let toolComponent = toolbar.toJSON();
  expect(toolComponent).toMatchSnapshot();
});