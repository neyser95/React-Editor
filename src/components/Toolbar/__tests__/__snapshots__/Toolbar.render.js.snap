// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar will render will all the underline and italic buttons passed as props 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    <i
      className="fas fa-underline"
    />
  </button>
  <button
    onClick={[Function]}
  >
    <i
      className="fas fa-italic"
    />
  </button>
</div>
`;
